### PR TITLE
Feature: interop tests

### PR DESCRIPTION
**Pull request by mprorock**
This is the beginnings of core interop tests for the trace profile.  All APIs and configs will be tested with Newman/Postman, and additional settings (security, etc) may be tested with other core system utilities and or JS

Initial testing will validate conformance with the interop profile for traceability as defined in this spec, and "happy path" on apis to start.  Once this is functional, and integrated into CI, we can add negative case and error case testing and handling, especially to include data integrity testing.  Tests are setup for logging to json, html, and others, and test names have been defined for parsing by analytics packages at a later state

Core postman collections for testing on interop that we will begin with are:

- service-providers.json : does the service provider meet baseline functionality
- reference-credentials.json : can the service provider issue, verify, etc with a known set of good credentials
- interop-credentials.json : can the service provider act in all required roles; 1) issuer, 2) verifier, 3) holder, with a known set of credentials from the trace vocab

I have also included some example postman tests as well as data packages to ensure that those unfamiliar with this type of testing approach can reference tests that are less complex to understand the basic test flow.

Will convert from draft once all tests are in and validated.  

Auth is disabled at this time, but these tests are configured to allow OAuth Bearer tokens for auth that can be re-enabled with one line.  Once we have sorted out issue #34 we can enable authZ/N as appropriate and integrate into CI

### CODE REVIEW

File: tests/postman/README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+The basic test flow is outlined in the below diagram:
+![Trace Interop Test Flow](./interop-test-flow.png)
+
+
+This test suite provides core postman collections for testing on interop that we will begin with are:
+
+- `service-providers.json` : does the service provider meet baseline functionality
+- `reference-credentials.json` : can the service provider issue, verify, etc with a known set of good credentials
+- `interop-credentials.json` : can the service provider act in all required roles; 1) issuer, 2) verifier, 3) holder, with a known set of credentials from the trace vocab
```
Comment (TallTed) [inline tests/postman/README.md:35]:
```suggestion
- `reference-credentials.json` : can the service provider issue, verify, etc., with a known set of good credentials
- `interop-credentials.json` : can the service provider act in all required roles -- 1) issuer, 2) verifier, 3) holder -- with a known set of credentials from the trace vocab
```

File: tests/postman/README.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+
+## Base Service Provider Config and Profile Tests
+
+To execute the base service provider tests from CLI execute the following:
```
Comment (TallTed) [inline tests/postman/README.md:40]:
```suggestion
To execute the base service provider tests from CLI, execute the following:
```

**Review by TallTed — CHANGES_REQUESTED** [review]
Typos, punctuation, etc.

**Comment by TallTed** [comment]
* Noticed during code skim: baseURL values are bare hostnames, lacking scheme -- so not URLs.  You mentioned forcing `https` somewhere, and that there would be errors if people tried to use http.  If that "force" doesn't _over-ride_ `http` or `ftp` or other scheme, it's not much of a force.  Also, better to raise an error and exit if someone's configured a bad scheme (`htpts`, anyone?)...

* Might be good to put `https://host-a.example.com` or similar in anything that's meant to be run elsewhere -- so you don't inadvertently launch a DDoS on yourself when everybody starts their testing with the values you've put in.

**Comment by mprorock** [comment]
so that this doesn't get too large I am going to convert this to a normal PR from draft, and after we get this accepted will add the final NxN tests in a separate PR as those we will want to be able to review separately.  note: I am not a JS guy - so some of the fixture-ing in `index.js` around the tests is merely "ok" and could probably be significantly improved.  I tried to balance parallel execution and sequential to optimize run time and avoid any weird overlapping execution issues, though i expect we may find some issues in there

**Review by OR13 — APPROVED** [review]

**Comment by OR13** [comment]
Lets take this PR before #33

**Comment by mkhraisha** [comment]
@mprorock this PR will take a bit of time for me to review unfortunately. can we hold of on merging it next tuesday if i havent reviewed it by then?

I think since this is such a big foundational PR we really ought to make sure everyone is happy with it.

**Comment by mprorock** [comment]
> I think since this is such a big foundational PR we really ought to make sure everyone is happy with it.

big +1 

I have branched off of this to add the NxN on a separate PR because that will also take significant time to review

File: tests/postman/README.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+Options:
+  -V, --version                  output the version number
+  -s, --service <file>           use the specified service provider test collection (default: "./collections/service-providers.json")
+  -r, --reference <file>         use the specified reference VC test collection (default: "./collections/reference-credentials.json")
+  -sd, --servicedata <file>      use the specified service provider data collection (default: "./data/service-providers.json")
+  -rd, --referencedata <file>    use the specified reference VC data collection (default: "./data/reference-credentials.json")
+  -rd, --reportdir <folder>      use the specified service provider data collection (default: "./newman")
+  -t, --tests <all...>           use the specified tests, "none" is provided as an option for dev purposes (choices: "all", "service", "reference", "interop", "none", default: ["all"])
+  -d, --dids <key, web, all...>  use the specified did methods (default: ["key"])
+  -n, --names <all...>           test only the service provider as identified by name (default: ["all"])
```
Comment (TallTed) [inline tests/postman/README.md:41]: This probably means changing the executable's source, wherever that is. It's to 
make the `-h` output more readable on typical terminal screens
```suggestion
  -s, --service <file>           use the specified service provider test collection 
                                 (default: "./collections/service-providers.json")
  -r, --reference <file>         use the specified reference VC test collection 
                                 (default: "./collections/reference-credentials.json")
  -sd, --servicedata <file>      use the specified service provider data collection 
                                 (default: "./data/service-providers.json")
  -rd, --referencedata <file>    use the specified reference VC data collection 
                                 (default: "./data/reference-credentials.json")
  -rd, --reportdir <folder>      use the specified service provider data collection 
                                 (default: "./newman")
  -t, --tests <all...>           use the specified tests, "none" is provided as an 
                                 option for dev purposes (choices: "all", "service", 
                                 "reference", "interop", "none", default: ["all"])
  -d, --dids <key, web, all...>  use the specified did methods (default: ["key"])
  -n, --names <all...>           test only the service provider as identified by name 
                                 (default: ["all"])
```

**Comment by mprorock** [comment]
per trace call - assuming that there is approaval from @mkhraisha with no substantive changes

File: tests/postman/README.md (hunk 3 above)
Comment (mprorock) [reply tests/postman/README.md:41]: I'll file an upstream issue with the cli lib - good catch - definitely better ways to optimize the output

**Review by mkhraisha — APPROVED** [review]

**Pull request merged**